viadot/sources/mindful.py

```diff
@@ -0,0 +1,251 @@
+        }
+
+        if endpoint == "surveys":
+            del params["start_date"]
+            del params["end_date"]
+        else:
+            start_date, end_date = self._check_timestamp(
+                start_date, end_date, date_interval
+            )
+            params["start_date"] = start_date
+            params["end_date"] = end_date
+
+        response = self._mindful_api_response(
+            endpoint=endpoint,
+            params=params,
+        )
+
+        if response.status_code == 200:
+            self.logger.info(
+                f"Succesfully downloaded {endpoint} data from mindful API."
+            )
+            data = StringIO(response.content.decode("utf-8"))
+        elif response.status_code == 204 and not response.content.decode():
+            self.logger.warning(
+                f"Thera are not {endpoint} data to download from {start_date or ' --- '} to {end_date  or ' --- '}."
+            )
+            data = json.dumps({})
+        else:
+            self.logger.error(
+                f"Failed to downloaded {endpoint} data. - {response.content}"
+            )
+            raise APIError(f"Failed to downloaded {endpoint} data.")
+
+        data_frame = pd.read_json(data)
+        if data_frame.empty:
+            self._handle_if_empty(
+                if_empty="warn",
+                message=f"No data was got from the endpoint: {endpoint}, days from {start_date} to {end_date}",
+            )
+
+        return data_frame
+
+    def to_file(
+        self,
+        data_frame: pd.DataFrame,
+        file_path: str = "",
+        file_name: Optional[str] = None,
+        file_extension: Literal["parquet", "csv"] = "csv",
+        sep: str = "\t",
+    ) -> str:
+        """Save Mindful response data to file and return filename.
+
+        Args:
+            data_frame (pd.DataFrame):
+            file_path (str, optional): Path where to save the file locally. Defaults to ''.
+            file_name (str, optional): Name of the file without extension. Defaults to None.
+            file_extension (Literal[parquet, csv], optional): File extensions for storing responses. Defaults to "csv".
+            sep (str, optional): Separator in csv file. Defaults to "\t".
+
+        returns
+            str: the absolute path of the downloaded file.
+        """
+
+        if file_name is None:
+            complete_file_name = f"mindful_response_{datetime.now().strftime('%Y%m%d%H%M%S')}.{file_extension}"
+            relative_path = os.path.join(file_path, complete_file_name)
+        else:
+            complete_file_name = f"{file_name}.{file_extension}"
+            relative_path = os.path.join(file_path, complete_file_name)
+
+        if file_extension == "csv":
+            data_frame.to_csv(relative_path, index=False, sep=sep)
+        elif file_extension == "parquet":
+            data_frame.to_parquet(relative_path, index=False)
+        else:
+            self.logger.warning(
```

Comment:
This should be a `ValueError` and verified at the top of the method, eg.
```python
if file_extension not in ["csv", "parquet"]:
    raise ValueError("`file_extension` must be one of: 'csv', 'parquet'.")
```

Reply:
I added the exception to the `if`-`else` sentence instead of adding a verification step.